app/src/main/java/org/xbmc/kore/jsonrpc/HostConnection.java

```diff
@@ -195,7 +207,7 @@ public HostConnection(final HostInfo hostInfo, int connectTimeout) {
         // Start with the default host protocol
         this.protocol = hostInfo.getProtocol();
         // Create a single threaded executor
-        this.executorService = Executors.newSingleThreadExecutor();
+        this.executorService = Executors.newFixedThreadPool(10);
```

Comment:
This might have unintended consequences. Having a single thread guaranteed that calls to Kodi were sequential, which is not guaranteed now.
I think that one example of this is happening in `PlaylistFragment`, when you choose the `Remove` action of an item in its overflow menu. That calls `Remove` on Kodi, doesn't care about the result, and immediately calls the code to refresh the playlist. With a single thread this would work because the `GetPlaylists` would only run after `Remove`, now that's not true, and it's causing the item not to be removed on the UI...
If this change stays, we need to review every single call to Kodi that doesn't care about the result to make sure that another call doesn't immediately follows, and expects the previous to have run.

Reply:
I think I fixed any possible racing condition regarding the playlists by reducing the calls to `refreshPlaylist` to a single call when the playlist has changed. We now fully depend on observing playlist notifications to handle playlists.